internal/component/prometheus/mapping/mapping.go

```diff
@@ -0,0 +1,224 @@
+package mapping
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/prometheus"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/internal/service/labelstore"
+	"github.com/grafana/alloy/internal/service/livedebugging"
+	prometheus_client "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/metadata"
+	"github.com/prometheus/prometheus/storage"
+	"go.uber.org/atomic"
+)
+
+const name = "prometheus.mapping"
+
+func init() {
+	component.Register(component.Registration{
+		Name:      name,
+		Stability: featuregate.StabilityGenerallyAvailable,
```

Comment:
We should mark this as PublicPreview for first release.
```suggestion
		Stability: featuregate.StabilityPublicPreview,
```